tests/integration_tests/functional/test_cpu_features.py

```diff
@@ -146,9 +146,33 @@ def test_cpu_template(test_microvm_with_ssh, network_config, cpu_template):
     containing all features on a t2/c3 instance and check that the cpuid in
     the guest is an exact match of the template.
     """
-    common_masked_features = ["avx512", "mpx", "clflushopt", "clwb", "xsavec",
-                              "xgetbv1", "xsaves", "pku", "ospke"]
-    c3_masked_features = ["avx2"]
+    common_masked_features_lscpu = ["dtes64", "monitor", "ds_cpl", "tm2",
+                                    "cnxt-id", "sdbg", "xtpr", "pdcm",
+                                    "osxsave",
+                                    "psn", "ds", "acpi", "tm", "ss", "pbe",
+                                    "fpdp", "rdt_m", "rdt_a", "mpx", "avx512f",
+                                    "intel_pt",
+                                    "avx512_vpopcntdq",
+                                    "3dnowprefetch", "pdpe1gb"]
+
+    common_masked_features_cpuid = {"SGX": "false", "HLE": "false",
+                                    "RTM": "false", "RDSEED": "false",
+                                    "ADX": "false", "AVX512IFMA": "false",
+                                    "CLFLUSHOPT": "false", "CLWB": "false",
+                                    "AVX512PF": "false", "AVX512ER": "false",
+                                    "AVX512CD": "false", "SHA": "false",
+                                    "AVX512BW": "false", "AVX512VL": "false",
+                                    "AVX512VBMI": "false", "PKU": "false",
+                                    "OSPKE": "false", "RDPID": "false",
+                                    "SGX_LC": "false",
+                                    "AVX512_4VNNIW": "false",
+                                    "AVX512_4FMAPS": "false",
+                                    "XSAVEC": "false", "XGETBV": "false",
+                                    "XSAVES": "false"}
+
+    # These are all discoverable by cpuid -1.
+    c3_masked_features = {"FMA": "false", "MOVBE": "false", "BMI": "false",
+                          "AVX2": "false", "BMI2": "false", "INVPCID": "false"}
 
```

Comment:
This is what `check_guest_cpuid_output` works with and since I wanted to use an already existent auxiliary function for this, I made it into a key-value store.